frontend/model/state.js

```diff
@@ -214,22 +185,33 @@ const mutations = {
 }
 // https://vuex.vuejs.org/en/getters.html
 // https://vuex.vuejs.org/en/modules.html
-//
-// !!  IMPORTANT  !!
-//
-// To make it simple to use *SOME* of these getters (the ones related to
-// group contract state) via 'state/groupContractSafeGetters', prefer using
-// 'getters' to access state instead of 'state'.
-//
-// For example, instead of: state[state.currentGroupId]
-//     Use getters instead: getters.currentGroupState
 const getters = {
+  // !!  IMPORTANT  !!
+  //
+  // For getters that get data from only contract state, write them
+  // under the 'getters' key of the object passed to DefineContract.
+  // See for example: frontend/model/contracts/group.js
+  //
+  // For convenience, we've defined the same getter, `currentGroupState`,
+  // twice, so that we can reuse the same getter definitions both here with Vuex,
+  // and inside of the contracts (e.g. in group.js).
+  //
+  // The one here is based off the value of `state.currentGroupId` — a user
+  // preference that does not exist in the group contract state.
+  //
+  // The getters in DefineContract are designed to be compatible with Vuex!
+  // When they're used in the context of DefineContract, their 'state' always refers
+  // to the state of the contract whose messages are being processed, regardless
+  // of what group we're in. That is why the definition of 'currentGroupState' in
+  // group.js simply returns the state.
+  //
+  // Since the getter functions are compatible between Vuex and our contract chain
+  // library, we can simply import them here, while excluding the getter for
+  // `currentGroupState`, and redefining it here based on the Vuex rootState.
+  ..._.omit(sbp('gi.contracts/group/getters'), ['currentGroupState']),
```

Comment:
What is this `.omit()` method?

Reply:
Oh nevermind, this is calling lodash! `..._.` was a really tricky "morse code" combination here 😅